server/src/main/java/org/opensearch/index/fielddata/ordinals/MultiOrdinals.java

```diff
@@ -213,13 +214,14 @@ public long getValueCount() {
         public boolean advanceExact(int docId) throws IOException {
             currentOffset = docId != 0 ? endOffsets.get(docId - 1) : 0;
             currentEndOffset = endOffsets.get(docId);
-            return currentOffset != currentEndOffset;
+            docValueCount = Math.toIntExact(currentEndOffset - currentOffset);
```

Comment:
I am assuming that `Math.toIntExact(` is to prevent any overflow? I am wondering what exactly it means if `currentEndOffset - currentOffset` is greater than MAX_INT?

Reply:
Yeah -- `toIntExact()` throws an `ArithmeticException` if it would overflow. 

In this case, I just moved it from the `docValueCount()` method. In practice, I don't think it's possible that we would ever hit this overflow. But because it's `long` arithmetic being adapted to a method that needs to return `int`, it's not a bad idea to check. Technically, by moving into the `advanceExact` method, I'm reducing the number of calls, since most code will call `docValueCount()` on every iteration.